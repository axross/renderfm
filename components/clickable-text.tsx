import React from "react";
import { twMerge } from "tailwind-merge";
import { primaryFocusRing } from "~/constants/style";

const clickableTextCss = twMerge(
  primaryFocusRing,
  "rounded-sm text-yellow-600 dark:text-yellow-400 hover:underline"
);

export interface ClickableTextProps {
  onClick?: React.MouseEventHandler<HTMLSpanElement>;
  className?: string;
  style?: React.CSSProperties;
  children?: React.ReactNode;
}

export const ClickableText: React.VFC<ClickableTextProps> = ({
  className,
  children,
  ...props
}) => {
  return (
    <span
      className={twMerge(clickableTextCss, "cursor-pointer", className)}
      tabIndex={0}
      {...props}
    >
      {children}
    </span>
  );
};

export type TextAnchorProps = React.AnchorHTMLAttributes<HTMLAnchorElement>;

export const TextAnchor: React.VFC<TextAnchorProps> = ({
  href,
  className,
  children,
  ...props
}) => {
  return (
    <a href={href} className={twMerge(clickableTextCss, className)} {...props}>
      {children}
    </a>
  );
};
