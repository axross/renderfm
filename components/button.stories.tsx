import * as React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";
import { Button } from "~/components/button";

export default {
  title: "Components/Button",
  component: Button,
  args: {
    variant: "primary",
    disabled: false,
    children: "Press Me",
  },
} as ComponentMeta<typeof Button>;

export const Example: ComponentStory<typeof Button> = (args) => (
  <Button {...args} />
);

export const Small = Example.bind({});
Small.args = {
  size: "sm",
};

export const ExtraSmall = Example.bind({});
ExtraSmall.args = {
  size: "xs",
};

export const Translucent = Example.bind({});
Translucent.args = {
  translucent: true,
};

export const Disabled = Example.bind({});
Disabled.args = {
  disabled: true,
};

export const Loading = Example.bind({});
Loading.args = {
  loading: true,
};
