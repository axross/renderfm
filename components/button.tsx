import { twMerge } from "tailwind-merge";
import * as React from "react";
import { LoadingIcon } from "~/components/icon";
import { primaryFocusRing } from "~/constants/style";

export type ButtonVariant = "primary";
export type ButtonSize = "base" | "sm" | "xs";

const DEFAULT_VARIANT: ButtonVariant = "primary";
const DEFAULT_SIZE: ButtonSize = "base";

export interface ButtonProps
  extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  variant?: ButtonVariant;
  size?: ButtonSize;
  translucent?: boolean;
  loading?: boolean;
}

export const Button: React.VFC<ButtonProps> = ({
  variant = DEFAULT_VARIANT,
  size = DEFAULT_SIZE,
  loading = false,
  translucent = false,
  disabled,
  className,
  children,
  ...props
}) => {
  return (
    <button
      disabled={disabled || loading}
      className={twMerge(
        "inline-flex items-center rounded-md disabled:cursor-not-allowed transition-all",
        variant === "primary"
          ? translucent
            ? "bg-yellow-600/20 hover:bg-yellow-600/30 disabled:bg-yellow-600/10 text-yellow-600"
            : "bg-yellow-600 disabled:bg-yellow-600/50 text-white"
          : null,
        variant === "primary" ? primaryFocusRing : null,
        size === "base" ? "px-4 py-2 text-base" : null,
        size === "sm" ? "px-3 py-1.5 text-base" : null,
        size === "xs" ? "px-2 py-1 text-sm" : null,
        className
      )}
      {...props}
    >
      {loading ? (
        <LoadingIcon
          className={twMerge(
            size === "base" ? "w-5 h-5 -ml-1.5 mr-2" : null,
            size === "sm" ? "w-4 h-4 -ml-1 mr-1.5" : null,
            size === "xs" ? "w-3 h-3 mr-1" : null
          )}
        />
      ) : null}

      {children}
    </button>
  );
};
