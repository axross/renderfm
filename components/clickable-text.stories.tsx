import { ComponentStory, ComponentMeta } from "@storybook/react";
import { ClickableText } from "~/components/clickable-text";

export default {
  title: "Components/ClickableText",
  component: ClickableText,
  args: {
    children: "This is a link",
  },
} as ComponentMeta<typeof ClickableText>;

export const Example: ComponentStory<typeof ClickableText> = (args) => (
  <ClickableText {...args} />
);
