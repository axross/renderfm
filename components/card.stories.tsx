import * as React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";
import { Button } from "~/components/button";
import { Card, CardFooter } from "~/components/card";

export default {
  title: "Components/Card",
  component: Card,
  args: {
    className: "w-96 min-h-64",
    children: "Hello",
  },
} as ComponentMeta<typeof Card>;

export const Example: ComponentStory<typeof Card> = (args) => (
  <Card {...args} />
);

export const WithFooter = Example.bind({});
WithFooter.args = {
  children: (
    <>
      Hello
      <CardFooter>
        <Button>Do Some Action</Button>
      </CardFooter>
    </>
  ),
};
