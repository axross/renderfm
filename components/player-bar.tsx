import { twMerge } from "tailwind-merge";
import { Button } from "./button";
import {
  CrossIcon,
  FastForwardIcon,
  LoadingIcon,
  PauseIcon,
  PlayIcon,
  RewindIcon,
} from "~/components/icon";
import { IconButton } from "~/components/icon-button";
import { SeekBar } from "~/components/seek-bar";

export interface PlayerBarProps {
  trackTitle: string;
  duration: number;
  currentTime: number;
  playing?: boolean;
  durationLoaded?: boolean;
  playbackLoading?: boolean;
  onResumeButtonClick?: () => void;
  onPauseButtonClick?: () => void;
  onRewindButtonClick?: () => void;
  onForwardButtonClick?: () => void;
  onCloseButtonClick?: () => void;
  onSeek?: (duration: number) => void;
  className?: string;
  style?: React.CSSProperties;
}

export const PlayerBar: React.VFC<PlayerBarProps> = ({
  trackTitle,
  duration,
  currentTime,
  playing = false,
  durationLoaded = true,
  playbackLoading = false,
  onResumeButtonClick,
  onPauseButtonClick,
  onRewindButtonClick,
  onForwardButtonClick,
  onCloseButtonClick,
  onSeek,
  className,
  ...props
}) => {
  return (
    <div
      className={twMerge("bg-slate-900 dark:bg-zinc-100", className)}
      {...props}
    >
      <div className="relative grid grid-cols-[auto_1fr_auto] pointer-fine:grid-cols-[1fr_auto] grid-rows-[auto_auto_auto] pointer-fine:grid-rows-[auto_auto] gap-x-4 items-start px-4 pt-6 pb-4 pointer-fine:pb-6">
        <div
          className="row-start-1 col-start-1 pointer-fine:col-start-2 self-end text-slate-100 dark:text-zinc-900 font-semibold truncate select-none"
          data-testid="track-title"
        >
          {trackTitle}
        </div>

        <div className="row-start-1 col-start-2 pointer-fine:col-start-3 self-end pr-6 text-slate-500 dark:text-zinc-500 text-xs font-mono select-none">
          <span data-testid="current-time">
            {durationLoaded ? formatDuration(currentTime) : "??:??"}
          </span>

          {"/"}

          <span data-testid="duration">
            {durationLoaded ? formatDuration(duration) : "??:??"}
          </span>
        </div>

        <SeekBar
          disabled={!durationLoaded}
          value={durationLoaded ? currentTime / duration : 0}
          onChange={onSeek}
          className="row-start-2 col-span-2 h-3 pointer-fine:h-2 mt-4 pointer-fine:mt-2"
          data-testid="seek-bar"
        />

        <div className="col-start-1 row-start-3 pointer-fine:row-start-1 row-span-2 pointer-fine:row-span-2 pointer-fine:self-center flex items-center justify-center gap-x-6 mt-6 pointer-fine:mt-0">
          <Button
            translucent
            disabled={playbackLoading}
            onClick={onRewindButtonClick}
            aria-busy={playbackLoading}
            className="pointer-fine:hidden"
            data-testid="rewind-button"
          >
            <RewindIcon />

            <span className="text-sm">15s</span>
          </Button>

          <IconButton
            size="lg"
            disabled={playbackLoading}
            onClick={playing ? onPauseButtonClick : onResumeButtonClick}
            className="focus:ring-offset-yellow-900 dark:focus:ring-offset-yellow-100"
            aria-busy={playbackLoading ? "true" : undefined}
            aria-label={playing ? "Pause" : "Play"}
            data-testid="playback-button"
          >
            {playbackLoading ? (
              <LoadingIcon />
            ) : playing ? (
              <PauseIcon />
            ) : (
              <PlayIcon />
            )}
          </IconButton>

          <Button
            translucent
            disabled={playbackLoading}
            onClick={onForwardButtonClick}
            aria-busy={playbackLoading}
            className="pointer-fine:hidden"
            data-testid="forward-button"
          >
            <span className="text-sm">15s</span>

            <FastForwardIcon />
          </Button>
        </div>

        <IconButton
          size="xs"
          translucent
          onClick={onCloseButtonClick}
          className="absolute top-2 right-2"
          data-testid="close-button"
        >
          <CrossIcon />
        </IconButton>
      </div>
    </div>
  );
};

function formatDuration(duration: number): string {
  const minutePart = Math.floor(duration / 60);
  const secondPart = `${Math.floor(duration % 60)}`.padStart(2, "0");

  return `${minutePart}:${secondPart}`;
}
