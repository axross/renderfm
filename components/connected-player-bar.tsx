import { twMerge } from "tailwind-merge";
import { PlayerBar } from "~/components/player-bar";
import { useHotkeyEffect } from "~/hooks/use-hotkey-effect";
import { useAudioPlayer } from "use-audio-player";
import React from "react";

interface ConnectedPlayerBar {
  className?: string;
  style?: React.CSSProperties;
}

export const ConnectedPlayerBar: React.VFC<ConnectedPlayerBar> = ({
  className,
  ...props
}) => {
  const player = useAudioPlayer();

  useHotkeyEffect(
    "ArrowLeft",
    () => {
      if (!player.isPlayable) return;

      player.seek(-15);
    },
    [player]
  );

  useHotkeyEffect(
    "ArrowRight",
    () => {
      if (!player.isPlayable) return;

      player.seek(15);
    },
    [player]
  );

  useHotkeyEffect(
    " ",
    () => {
      if (!player.isPlayable) return;

      if (player.isPlaying) {
        player.pause();
      } else {
        player.play();
      }
    },
    [player]
  );

  return (
    <PlayerBar
      trackTitle={player.track?.title ?? ""}
      duration={player.duration}
      currentTime={player.currentTime}
      playing={player.isPlaying}
      durationLoaded={player.isDurationLoaded}
      playbackLoading={player.isPlaybackLoading}
      onPauseButtonClick={player.pause}
      onResumeButtonClick={player.play}
      onRewindButtonClick={React.useCallback(() => player.seek(-15), [player])}
      onForwardButtonClick={React.useCallback(() => player.seek(15), [player])}
      onSeek={(d) => player.seekTo(player.duration * d)}
      onCloseButtonClick={player.unload}
      className={twMerge(
        "fixed left-0 right-0 transition-all",
        player.track !== null
          ? "bottom-0"
          : "-bottom-40 pointer-fine:-bottom-24",
        className
      )}
      {...props}
    />
  );
};
