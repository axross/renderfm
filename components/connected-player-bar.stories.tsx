import * as React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";
import { Button } from "~/components/button";
import { ConnectedPlayerBar } from "~/components/connected-player-bar";
import {
  AudioPlayerProvider,
  createBrowserAudioPlayer,
  useAudioPlayer,
} from "use-audio-player";

export default {
  title: "Connected Components/PlayerBar",
  component: ConnectedPlayerBar,
  decorators: [
    (Story) => {
      const player = createBrowserAudioPlayer();

      return (
        <AudioPlayerProvider player={player}>
          <Story />
        </AudioPlayerProvider>
      );
    },
  ],
} as ComponentMeta<typeof ConnectedPlayerBar>;

export const Example: ComponentStory<typeof ConnectedPlayerBar> = (args) => {
  const player = useAudioPlayer();

  return (
    <>
      <Button
        onClick={() => {
          player.load({
            title: "Some Fancy Music",
            src: "https://file-examples-com.github.io/uploads/2017/11/file_example_MP3_5MG.mp3",
          });

          // eslint-disable-next-line storybook/await-interactions, storybook/context-in-play-function
          player.play();
        }}
      >
        Play Something
      </Button>

      <ConnectedPlayerBar {...args} />
    </>
  );
};
