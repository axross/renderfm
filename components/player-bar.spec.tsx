// eslint-disable-next-line no-var
var MockSeekBar: jest.Mock;
jest.mock("~/components/seek-bar", () => {
  MockSeekBar = jest.fn();

  return {
    SeekBar: MockSeekBar,
  };
});

import { randNumber, randPhrase } from "@ngneat/falso";
import { render, screen, within } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { PlayerBar, PlayerBarProps } from "~/components/player-bar";
import type { SeekBarProps } from "~/components/seek-bar";

describe("<PlayerBar>", () => {
  function setup(additionalProps?: Partial<PlayerBarProps>) {
    const onResumeButtonClick = jest.fn();
    const onPauseButtonClick = jest.fn();
    const onRewindButtonClick = jest.fn();
    const onForwardButtonClick = jest.fn();
    const onCloseButtonClick = jest.fn();
    const onSeek = jest.fn();

    render(
      <PlayerBar
        trackTitle={randPhrase()}
        duration={randNumber({ min: 500, max: 1000 })}
        currentTime={randNumber({ min: 300, max: 500 })}
        onResumeButtonClick={onResumeButtonClick}
        onPauseButtonClick={onPauseButtonClick}
        onRewindButtonClick={onRewindButtonClick}
        onForwardButtonClick={onForwardButtonClick}
        onCloseButtonClick={onCloseButtonClick}
        onSeek={onSeek}
        data-testid="player-bar"
        {...additionalProps}
      />
    );

    const playerBar = () => screen.getByTestId("player-bar");
    const playbackButton = () =>
      within(playerBar()).getByTestId("playback-button");
    const rewindButton = () => within(playerBar()).getByTestId("rewind-button");
    const forwardButton = () =>
      within(playerBar()).getByTestId("forward-button");
    const closeButton = () => within(playerBar()).getByTestId("close-button");

    return {
      playerBar,
      trackTitle: () => within(playerBar()).getByTestId("track-title"),
      played: () => within(playerBar()).getByTestId("current-time"),
      duration: () => within(playerBar()).getByTestId("duration"),
      playbackButton,
      rewindButton,
      forwardButton,
      closeButton,
      seekBar: () => within(playerBar()).getByTestId("seek-bar"),
      onResumeButtonClick,
      onPauseButtonClick,
      onRewindButtonClick,
      onForwardButtonClick,
      onCloseButtonClick,
      onSeek,
      clickPlaybackButton: () => {
        userEvent.click(playbackButton());
      },
      clickRewindButton: () => {
        userEvent.click(rewindButton());
      },
      clickForwardButton: () => {
        userEvent.click(forwardButton());
      },
      clickCloseButton: () => {
        userEvent.click(closeButton());
      },
      emitSeekBarOnChange: (
        value: Parameters<Required<SeekBarProps>["onChange"]>[0]
      ) => {
        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
        seekBarOnChange!(value);
      },
    };
  }

  const setupPlaying: typeof setup = (additionalProps) =>
    setup({ ...additionalProps, playing: true });

  const setupPlaybackLoading: typeof setup = (additionalProps) =>
    setup({ ...additionalProps, playbackLoading: true });

  let seekBarOnChange: SeekBarProps["onChange"];

  beforeEach(() => {
    MockSeekBar.mockImplementation((props) => {
      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access
      seekBarOnChange = props.onChange;

      return <span data-mockfor="SeekBar" {...props} />;
    });
  });

  it("renders the player bar that displays episode title, duration and several controls", () => {
    const {
      trackTitle,
      played,
      duration,
      rewindButton,
      forwardButton,
      closeButton,
      seekBar,
    } = setup();

    expect(trackTitle()).toMatchSnapshot();
    expect(played()).toMatchSnapshot();
    expect(duration()).toMatchSnapshot();
    expect(rewindButton()).toMatchSnapshot();
    expect(forwardButton()).toMatchSnapshot();
    expect(closeButton()).toMatchSnapshot();
    expect(seekBar()).toMatchSnapshot();
  });

  it("renders the player bar that contains play button", () => {
    const { playbackButton } = setup();

    expect(playbackButton()).toMatchSnapshot();
  });

  it("renders the player bar that contains pause button when playing=true is given", () => {
    const { playbackButton } = setupPlaying();

    expect(playbackButton()).toMatchSnapshot();
  });

  it("renders the play/pause button, rewind button and forward button disabled when playbackLoading=true is given", () => {
    const { rewindButton, forwardButton } = setupPlaybackLoading();

    expect(rewindButton()).toBeDisabled();
    expect(forwardButton()).toBeDisabled();
  });

  it("calls onResumeButtonClick() when the user clicks the play button", () => {
    const { onResumeButtonClick, clickPlaybackButton } = setup();

    clickPlaybackButton();

    expect(onResumeButtonClick).toHaveBeenCalled();
  });

  it("calls onPauseButtonClick() when the user clicks the pause button", () => {
    const { onPauseButtonClick, clickPlaybackButton } = setupPlaying();

    clickPlaybackButton();

    expect(onPauseButtonClick).toHaveBeenCalled();
  });

  it("calls onRewindButtonClick() when the user clicks the rewind button", () => {
    const { onRewindButtonClick, clickRewindButton } = setup();

    clickRewindButton();

    expect(onRewindButtonClick).toHaveBeenCalled();
  });

  it("calls onForwardButtonClick() when the user clicks the forward button", () => {
    const { onForwardButtonClick, clickForwardButton } = setup();

    clickForwardButton();

    expect(onForwardButtonClick).toHaveBeenCalled();
  });

  it("calls onCloseButtonClick() when the user clicks the close button", () => {
    const { onCloseButtonClick, clickCloseButton } = setup();

    clickCloseButton();

    expect(onCloseButtonClick).toHaveBeenCalled();
  });

  it("calls onSeek() when the seek bar's onChange gets called", () => {
    const { onSeek, emitSeekBarOnChange } = setup();

    emitSeekBarOnChange(0.5);
    emitSeekBarOnChange(0.25);
    emitSeekBarOnChange(1);

    expect(onSeek).toHaveBeenCalledTimes(3);
    expect(onSeek).toHaveBeenNthCalledWith(1, 0.5);
    expect(onSeek).toHaveBeenNthCalledWith(2, 0.25);
    expect(onSeek).toHaveBeenNthCalledWith(3, 1);
  });
});
