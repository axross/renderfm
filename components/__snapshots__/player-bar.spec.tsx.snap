// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlayerBar> renders the player bar that contains pause button when playing=true is given 1`] = `
<button
  aria-label="Pause"
  class="inline-flex items-center rounded-md disabled:cursor-not-allowed bg-yellow-600 disabled:bg-yellow-600/50 text-white outline-none focus:ring-4 focus:ring-yellow-500/50 transition-all p-2 focus:ring-offset-yellow-900 dark:focus:ring-offset-yellow-100"
  data-testid="playback-button"
>
  <svg
    class="w-8 h-8"
    fill="none"
    stroke="currentColor"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M10 9v6m4-6v6m7-3a9 9 0 11-18 0 9 9 0 0118 0z"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="1.5"
    />
  </svg>
</button>
`;

exports[`<PlayerBar> renders the player bar that contains play button 1`] = `
<button
  aria-label="Play"
  class="inline-flex items-center rounded-md disabled:cursor-not-allowed bg-yellow-600 disabled:bg-yellow-600/50 text-white outline-none focus:ring-4 focus:ring-yellow-500/50 transition-all p-2 focus:ring-offset-yellow-900 dark:focus:ring-offset-yellow-100"
  data-testid="playback-button"
>
  <svg
    class="w-8 h-8"
    fill="none"
    stroke="currentColor"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M14.752 11.168l-3.197-2.132A1 1 0 0010 9.87v4.263a1 1 0 001.555.832l3.197-2.132a1 1 0 000-1.664z"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="1.5"
    />
    <path
      d="M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="1.5"
    />
  </svg>
</button>
`;

exports[`<PlayerBar> renders the player bar that displays episode title, duration and several controls 1`] = `
<div
  class="row-start-1 col-start-1 pointer-fine:col-start-2 self-end text-slate-100 dark:text-zinc-900 font-semibold truncate select-none"
  data-testid="track-title"
>
  navigating the hard drive wont do anything, we need to synthesize the auxiliary USB circuit!
</div>
`;

exports[`<PlayerBar> renders the player bar that displays episode title, duration and several controls 2`] = `
<span
  data-testid="current-time"
>
  7:00
</span>
`;

exports[`<PlayerBar> renders the player bar that displays episode title, duration and several controls 3`] = `
<span
  data-testid="duration"
>
  16:01
</span>
`;

exports[`<PlayerBar> renders the player bar that displays episode title, duration and several controls 4`] = `
<button
  aria-busy="false"
  class="inline-flex items-center rounded-md disabled:cursor-not-allowed bg-yellow-600/20 hover:bg-yellow-600/30 disabled:bg-yellow-600/10 text-yellow-600 outline-none focus:ring-4 focus:ring-yellow-500/50 transition-all px-4 py-2 text-base pointer-fine:hidden"
  data-testid="rewind-button"
>
  <svg
    class="h-6 w-6"
    fill="none"
    stroke="currentColor"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M12.066 11.2a1 1 0 000 1.6l5.334 4A1 1 0 0019 16V8a1 1 0 00-1.6-.8l-5.333 4zM4.066 11.2a1 1 0 000 1.6l5.334 4A1 1 0 0011 16V8a1 1 0 00-1.6-.8l-5.334 4z"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="1.5"
    />
  </svg>
  <span
    class="text-sm"
  >
    15s
  </span>
</button>
`;

exports[`<PlayerBar> renders the player bar that displays episode title, duration and several controls 5`] = `
<button
  aria-busy="false"
  class="inline-flex items-center rounded-md disabled:cursor-not-allowed bg-yellow-600/20 hover:bg-yellow-600/30 disabled:bg-yellow-600/10 text-yellow-600 outline-none focus:ring-4 focus:ring-yellow-500/50 transition-all px-4 py-2 text-base pointer-fine:hidden"
  data-testid="forward-button"
>
  <span
    class="text-sm"
  >
    15s
  </span>
  <svg
    class="h-6 w-6"
    fill="none"
    stroke="currentColor"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M11.933 12.8a1 1 0 000-1.6L6.6 7.2A1 1 0 005 8v8a1 1 0 001.6.8l5.333-4zM19.933 12.8a1 1 0 000-1.6l-5.333-4A1 1 0 0013 8v8a1 1 0 001.6.8l5.333-4z"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="1.5"
    />
  </svg>
</button>
`;

exports[`<PlayerBar> renders the player bar that displays episode title, duration and several controls 6`] = `
<button
  class="inline-flex items-center rounded-md disabled:cursor-not-allowed bg-yellow-600/20 hover:bg-yellow-600/30 disabled:bg-yellow-600/10 text-yellow-600 outline-none focus:ring-4 focus:ring-yellow-500/50 transition-all p-1 absolute top-2 right-2"
  data-testid="close-button"
>
  <svg
    class="w-4 h-4"
    fill="none"
    stroke="currentColor"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M6 18L18 6M6 6l12 12"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="1.5"
    />
  </svg>
</button>
`;

exports[`<PlayerBar> renders the player bar that displays episode title, duration and several controls 7`] = `
<span
  class="row-start-2 col-span-2 h-3 pointer-fine:h-2 mt-4 pointer-fine:mt-2"
  data-mockfor="SeekBar"
  data-testid="seek-bar"
  value="0.43704474505723206"
/>
`;
